cmd/post.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright © 2022 himkt <himkt@klis.tsukuba.ac.jp>
+
+*/
+package cmd
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"path"
+
+	"github.com/spf13/cobra"
+)
+
+type Payload struct {
+	Content string `json:"content"`
+}
+
+// postCmd represents the post command
+var postCmd = &cobra.Command{
+	Use:   "post",
+	Short: "Post text to discord server",
+	Long:  `Post text to discord server using HTTP.`,
+	Run: func(cmd *cobra.Command, args []string) {
+		home, _ := os.UserHomeDir()
+		configPath := path.Join(home, ".config/dobato/webhook")
+		if _, err := os.Stat(configPath); errors.Is(err, os.ErrNotExist) {
+			fmt.Println("No config available, run dobato setup first.")
+			return
+		}
+
+		f, err := os.Open(configPath)
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+		}
+
+		defer f.Close()
+		scanner := bufio.NewScanner(f)
+
+		var webhookUrl string
+		if scanner.Scan() {
+			webhookUrl = scanner.Text()
+		} else {
+			panic("Failed to read webhook")
+		}
+
+		text, err := cmd.Flags().GetString("text")
+		if err != nil {
+			fmt.Fprintln(os.Stderr)
+			return
+		}
+
+		payload := new(Payload)
+		payload.Content = text
+
+		json, err := json.Marshal(payload)
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			return
+		}
+
+		fmt.Println("web: ", webhookUrl)
+		res, err := http.Post(webhookUrl, "application/json", bytes.NewBuffer(json))
```

Comment:
On the docker container, I observed the following error.

```
> docker run --rm -it --platform linux/amd64 ubuntu:20.04 bash
root@bc252b7cfa1f:/# ./dobato-linux-amd64 post --text "linux-amd64"

Post "https://discord.com/api/webhooks/...": x509: certificate signed by unknown authority
```

https://qiita.com/reikkk/items/e81fe384ad83a8e8b845 solved it.